playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -1108,10 +1213,16 @@ public void load(PKMediaSourceConfig mediaSourceConfig) {
         if (player == null) {
             this.useTextureView = playerSettings.useTextureView();
             this.isSurfaceSecured = playerSettings.isSurfaceSecured();
+            this.useSpeedAdjustingRenderer =
+                    mediaSourceConfig.mediaSource.getMediaFormat() == PKMediaFormat.udp
+                            && playerSettings.getMulticastSettings().getExperimentalAdjustSpeedOnNegativePosition();
```

Comment:
this is kind of repeating, in this line you check both conditions and in the else clause when player is already set and we switch to multicast you pass the first part of the argument to `maybeReInitPlayerOnSpeedAdjustmentChange` and then check it with `&&` inside with the 2nd part so extract 
```
useSpeedAdjustingRenderer =
                    mediaSourceConfig.mediaSource.getMediaFormat() == PKMediaFormat.udp
                            && playerSettings.getMulticastSettings().getExperimentalAdjustSpeedOnNegativePosition();
```
to a `shouldUseSpeedAdjustingRenderer` method and pass to `maybeReInitPlayerOnSpeedAdjustmentChange` just the format and call inside it to `shouldUseSpeedAdjustingRenderer` as well